clang/include/clang/Parse/Parser.h

```diff
@@ -488,6 +488,49 @@ class Parser : public CodeCompletionHandler {
   /// a statement expression and builds a suitable expression statement.
   StmtResult handleExprStmt(ExprResult E, ParsedStmtContext StmtCtx);
 
+  class AssumeParseAssociatedStmtRAII {
+    Parser *parent;
```

Comment:
Since I am creating instance of `AssumeParseAssociatedStmtRAII` in `Parser::ParseOpenMPAssumesDirective()` I have to store Parent as a pointer. The creation of the instance is done as :
`AssumeParseAssociatedStmtRAII AssumeParseAssocRAII(this, Loc, DKind);`

Reply:
Pass it as *this